plugins/recall/scripts/session-index.py

```
@@ -0,0 +1,565 @@
+
+    # ⚠ Refuse to prune on an empty scan. glob() SWALLOWS I/O errors and
+    # returns [], so an unreadable transcripts dir — an NFS/mutagen hiccup, a
+    # permissions blip, a home not yet mounted — made every indexed path look
+    # deleted and wiped the whole index, reporting {"removed": N, "failed": 0}
+    # as success. The hook discards that output, so the first symptom would be
+    # /recall answering "no transcripts indexed". The read path already refuses
+    # to repair by deletion; the write path owes the same refusal.
+    if not files and existing:
+        raise IndexError_(
+            "found no transcripts under %s but the index holds %d — refusing to "
+            "prune. Check the directory is readable." % (PROJECTS_DIR, len(existing))
```

Ruff (0.16.1)

[warning] Use format specifiers instead of percent format

(UP031)